[BROWSEUI] Show error on invalid path (CORE-10989)

giawa opened the pull request:
## Purpose

This PR displays an error message when the path entered into the address bar is non-existent.

JIRA issue: [CORE-10989](https://jira.reactos.org/browse/CORE-10989)

## Proposed changes

- Add localization .rc files for the new error message and title
- Display an error message if shell32 returns either ERROR_INVALID_DRIVE or ERROR_FILE_NOT_FOUND when parsing the path name

## TODO

Shell32 does not currently return ERROR_INVALID_DRIVE when the path is a drive that doesn't exist.  I believe this is a shortfall of the ReactOS/Wine implementation of shell32, and will be 'fixed' once shell32 is patched.  I'll look into why shell32 is not returning this error code, and may be able to patch it in the future.

## Screenshot

![invalidpath](https://user-images.githubusercontent.com/3923687/79698233-78874100-823c-11ea-93a8-c14a07ad6961.png)

learn-more commented on dll/win32/browseui/addresseditbox.cpp:
```diff
@@ -173,6 +173,21 @@ HRESULT STDMETHODCALLTYPE CAddressEditBox::ParseNow(long paramC)
 
+HRESULT STDMETHODCALLTYPE CAddressEditBox::ShowFileNotFoundError(HRESULT hRet)
+{
+    LPWSTR input;
+    int inputLength = fCombobox.GetWindowTextLength() + 2;
+
+    input = new WCHAR[inputLength];
+    fCombobox.GetWindowText(input, inputLength);
+
+    ShellMessageBoxW(_AtlBaseModule.GetResourceInstance(), fCombobox.m_hWnd, MAKEINTRESOURCEW(IDS_PARSE_ADDR_ERR_TEXT), MAKEINTRESOURCEW(IDS_PARSE_ADDR_ERR_TITLE), MB_OK | MB_ICONERROR, input);
+
+    delete[] input;
+
+    return hRet;
```
When using a CComHeapPtr like this, you do not have to worry about cleanup:
```suggestion
    CComHeapPtr<WCHAR> input;
    int inputLength = fCombobox.GetWindowTextLength() + 2;

    input.Allocate(inputLength);
    fCombobox.GetWindowText(input, inputLength);

    ShellMessageBoxW(_AtlBaseModule.GetResourceInstance(), fCombobox.m_hWnd, MAKEINTRESOURCEW(IDS_PARSE_ADDR_ERR_TEXT), MAKEINTRESOURCEW(IDS_PARSE_ADDR_ERR_TITLE), MB_OK | MB_ICONERROR, input);

    return hRet;
```

(lgtm either way, just something to keep in mind :) )

giawa replied:
I like it!  However, I wasn't able to find `input.GetString()`, but `input.m_pData` worked.  Is that expected?

learn-more replied:
My bad, I typed this in the github interface. (`GetString` is from CAtlString, which you can also use here btw)...

For `CComHeapPtr` (or `CHeapPtr` would also work..) `m_pData` is fine, or you can cast it.
Normally using that is not needed when passing it to functions that accept strings,
but since we are passing it to a vararg function, the compiler cannot know what we want without making it explicit.

giawa replied:
Sounds good, I'll alter the commit to use CComHeapPtr

HBelusca commented on dll/win32/browseui/lang/fr-FR.rc:
```diff
@@ -354,3 +354,9 @@ BEGIN
+
+STRINGTABLE
+BEGIN
+    IDS_PARSE_ADDR_ERR_TITLE "File Explorer"
```
```suggestion
    IDS_PARSE_ADDR_ERR_TITLE "Explorateur de fichiers"
```

HBelusca replied:
Note also (in all the translations) that we could replace "File Explorer" by "ReactOS Explorer" since this is denomination can be seen around:
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=%28%5Cs%29explorer&sr=1

giawa replied:
Would you like me to also replace the french version with "ReactOS Explorer"?  Or keep the translated "Explorateur de fichiers"?

HBelusca replied:
It all depends on what you decide to do for the msgbox error title :)
If you decide to use "ReactOS Explorer", then yes for french it would be "Explorateur ReactOS".

giawa replied:
Went for ReactOS Explorer in the new commit.

HBelusca commented on dll/win32/browseui/lang/fr-FR.rc:
```diff
@@ -354,3 +354,9 @@ BEGIN
+STRINGTABLE
+BEGIN
+    IDS_PARSE_ADDR_ERR_TITLE "File Explorer"
+    IDS_PARSE_ADDR_ERR_TEXT "File Explorer can't find '%1'. Check the spelling and try again."
```
```suggestion
    IDS_PARSE_ADDR_ERR_TEXT "L'explorateur de fichiers ne peut trouver '%1'. Veuillez vérifier l'orthographe et réessayer."
```

HBelusca replied:
Also in english, "can't" doesn't look too much professional. "cannot" would be better.

giawa replied:
Fair enough - I took similar verbiage from win10, but xp does use "cannot".

HBelusca replied:
Windows <= 2003 use the "pro" verbiage; vista+ use the "chad" version LMAO :sweat_smile: :sweat_smile:

giawa replied:
Went for 'pro'.

giawa commented:
Updated commit to use 'ReactOS Explorer', and to use `CComHeapPtr` instead of new/delete.  Also updated some of the verbiage in the resource files.

![invalidpath](https://user-images.githubusercontent.com/3923687/79699815-b12c1800-8246-11ea-8a34-5115460839c8.png)

KyleKatarn31 commented on dll/win32/browseui/lang/fr-FR.rc:
```diff
@@ -354,3 +354,9 @@ BEGIN
+
+STRINGTABLE
+BEGIN
+    IDS_PARSE_ADDR_ERR_TITLE "Explorateur ReactOS"
```
Current windows text is "Explorateur de fichiers"

giawa replied:
We decided above to use 'ReactOS Explorer', which would translate to 'L'explorateur ReactOS' (commend from HBelusca above).

KyleKatarn31 commented on dll/win32/browseui/lang/fr-FR.rc:
```diff
@@ -354,3 +354,9 @@ BEGIN
+STRINGTABLE
+BEGIN
+    IDS_PARSE_ADDR_ERR_TITLE "Explorateur ReactOS"
+    IDS_PARSE_ADDR_ERR_TEXT "L'explorateur ReactOS ne peut trouver '%1'. Veuillez vérifier l'orthographe et réessayer."
```
Current windows text is "Windows ne parvient pas à trouver '%1'.  Vérifiez l'orthographe et réessayer."

giawa replied:
Either one works for me - the current commit uses a suggestion from HBelusca above.  I'd prefer to not get caught up in translation details and have future PRs submitted from translators to localize the resources.  Would that work?

KyleKatarn31 replied:
For the title : OK for your proposal
For the text : @HBelusca : would you agree on my suggestion ?

learn-more replied:
@giawa that is perfectly fine, feel free to not apply translations at all unless you want to.

HBelusca replied:
Agreed on my side.

KyleKatarn31 replied:
OK so @giawa, please update your proposal for  IDS_PARSE_ADDR_ERR_TEXT and let's go :)

giawa replied:
Sorry Kyle, got merged before I could push that change.  We'll have to do it via another PR.

KyleKatarn31 replied:
OK, I take care of this.
